engine/src/main/scala/cromwell/engine/db/DataAccess.scala

```diff
@@ -42,6 +44,50 @@ object DataAccess {
 
   val FailureEventMaxMessageLength = 1024
   val RetryBackoff = SimpleExponentialBackoff(50 millis, 1 seconds, 1D)
+
+  private lazy val WorkflowMetadataSummarySemigroup = new Semigroup[WorkflowMetadataSummary] {
+    override def append(summary1: WorkflowMetadataSummary,
+                        summary2: => WorkflowMetadataSummary): WorkflowMetadataSummary = {
+      // Resolve the status if both `this` and `that` have defined statuses.  This will evaluate to `None`
+      // if one or both of the statuses is not defined.
+      val resolvedStatus = for {
+        thisStatus <- summary1.status map WorkflowState.fromString
+        thatStatus <- summary2.status map WorkflowState.fromString
+      } yield (thisStatus |+| thatStatus).toString
+
+      WorkflowMetadataSummary(
+        workflowUuid = summary1.workflowUuid,
+        // If not both statuses are defined, take whichever is defined.
+        status = resolvedStatus orElse summary1.status orElse summary2.status,
+        name = summary1.name orElse summary2.name,
+        startDate = summary1.startDate orElse summary2.startDate,
+        endDate = summary1.endDate orElse summary2.endDate
+      )
+    }
+  }
+
+  private def getUpdatedSummary(existingSummary: Option[WorkflowMetadataSummary],
+                                metadataForUuid: Seq[Metadatum]): WorkflowMetadataSummary = {
+    implicit val wmss = WorkflowMetadataSummarySemigroup
+
+    val baseSummary = existingSummary.getOrElse(WorkflowMetadataSummary(metadataForUuid.head.workflowUuid))
+    metadataForUuid.foldLeft(baseSummary) {
+      case (metadataSummary, metadatum) => metadataSummary |+| toSummary(metadatum)
+    }
+  }
+
+  private def toSummary(metadatum: Metadatum): WorkflowMetadataSummary = {
```

Comment:
It was only used in one place, so I changed it from a implicit class with a method to just a method. I'll put it back as the original, as this PR is targeting your branch.